internal/configs/configschema/coerce_value.go

```diff
@@ -78,7 +87,15 @@ func (b *Block) coerceValue(in cty.Value, path cty.Path) (cty.Value, error) {
 		attrs[name] = val
 	}
 
-	for typeName, blockS := range b.BlockTypes {
+	// Stable sort keys for consistent error reports
```

Comment:
I'm kind of split on this.

On the one hand, consistent error messages make sense.

On the other hand, this will require an effort on our side in all such places in the codebase (I expect there to be a bunch) and I'm not sure the (very limited imo) value is worth the effort.

On the other other hand, we could also use Diagnostics here instead of failing on the first error. Basically, before the for loop create a diagnostics struct, and then append the errors to it, continuing the loop. This way we'll get all the errors. But if this is the approach we wanna take, I'd put in an issue and delay it past the stable release.

Reply:
I see the value in collecting all error messages. It's especially useful when user validates the configuration using `tofu validate`.

Reply:
I think that wherever we are iterating through maps where there are user facing messages, we should try to sort if possible.  I see this as an ongoing cleanup when we encounter it.

As far as tfdialogs,  I think that's the right long term solution, but it might drastically baloon the scope of this bugfix.  I'll take a look into it this morning and give a rough estimate on how hard it would be.  Alternatively I could return a more complex error type as a stopgap.

Reply:
It's definitely a non-trivial amount of work to introduce a complex error type / diags and make sure functionality is not broken.  I'm going to recommend we keep the sort and add a ticket to the backlog.

Reply:
Created https://github.com/opentofu/opentofu/issues/881 to track the follow up work proposed here.

Reply:
The remaining question is should we back out the stable sort to be consistent or set precedence that we fix these when we encounter them, either with a stable sort or better use of tfdiags/complex errors

Reply:
> I think that wherever we are iterating through maps where there are user facing messages, we should try to sort if possible.

I don't care too much either way, so happy to do it the way you propose